### DISCUSSION TITLE
Python code performs worse than terminal command

### DISCUSSION THREAD

**Discussion by haakonjacobsen** [openai/whisper#519] [Q&A] [answered]
Hi, I just played around with whisper and found something a bit strange. When I run whisper in the terminal: 
```zsh
whisper "audio.mp3" --task transcribe --model base
```
it performs better than this simple python script:

```python
import whisper
from pathlib import Path
from whisper.utils import write_srt

def main():
    model = whisper.load_model("base")
    audio_path = "audio.mp3"
    output_dir = Path("subs")

    # transcribe the audio
    result = whisper.transcribe(model, audio_path)

    # save SRT
    audio_basename = Path(audio_path).stem
    language = result["language"]
    sub_name = f"{audio_basename}.{language}.srt"
    with open(Path(output_dir) / (sub_name), "w", encoding="utf-8") as srt:
        write_srt(result["segments"], file=srt)



if __name__ == "__main__":
    main()
```

By better, I mean that it's more accurate, just by reading over the two generated srt files.
I assumed that the output of the model would be the same since I used the same pre-trained **base** model, and I assumed that the output is deterministic when using the same models without specifying any of the options . Is there a difference in the base configurations when I run the python script instead of running it from the terminal? Does anyone know if I am doing anything wrong in the python script?

**Comment by jianfch** [comment, marked as answer]
They have different default settings. By default, terminal uses beam search which generally performs better than greedy search (python's default).  See #177.

**Reply by haakonjacobsen** [reply]
Thanks, that explains it!
